packages/paste-core/components/pagination/src/PaginationArrow.tsx

```diff
@@ -0,0 +1,107 @@
+import type {PaginationArrowProps} from './types';
+import {PaginationArrowPropTypes} from './proptypes';
+import {handleLinkValidation} from './utils';
+
+const PaginationArrow = React.forwardRef<HTMLButtonElement, PaginationArrowProps>(
+  (
+    {
+      as = 'button',
+      children,
+      direction = 'left',
+      href,
+      nextArrowLabel = 'Go to next page',
+      previousArrowLabel = 'Go to previous page',
+      ...props
+    },
+    ref
+  ) => {
+    handleLinkValidation({as, href});
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        ref={ref}
+        as={as}
+        href={href}
+        alignItems="center"
+        appearance="none"
+        background="none"
+        border="none"
+        borderColor="colorBorder"
+        borderRadius={children ? 'borderRadius10' : 'borderRadiusCircle'}
+        color="colorText"
+        display="flex"
+        fontFamily="fontFamilyText"
+        fontVariantNumeric="tabular-nums"
+        fontWeight="fontWeightNormal"
+        lineHeight="lineHeight30"
+        outline="none"
+        padding="space0"
+        position="relative"
+        textDecoration="none"
+        transition="border-color 100ms ease-in, box-shadow 100ms ease-in, color 100ms ease-in"
+        _hover={{borderColor: 'colorBorderPrimary', color: 'colorTextLink', cursor: 'pointer', textDecoration: 'none'}}
+        _focus={{
+          borderColor: 'colorBorderPrimary',
+          boxShadow: 'shadowFocus',
+          color: 'colorTextLink',
+          textDecoration: 'none',
+        }}
+        _active={{textDecoration: 'none'}}
+      >
+        {direction === 'left' ? (
+          <Box
+            alignItems="center"
+            borderColor="inherit"
+            borderRadius="borderRadiusCircle"
+            borderStyle="solid"
+            borderWidth="borderWidth10"
+            display="flex"
+            height="sizeSquare80"
+            justifyContent="center"
+            width="sizeSquare80"
+          >
+            <ArrowBackIcon decorative={false} display="block" title={previousArrowLabel} />
+          </Box>
+        ) : null}
+        {children ? (
+          <Text
+            as="span"
+            color="inherit"
+            marginLeft={direction === 'left' ? 'space30' : 'space0'}
+            marginRight={direction === 'right' ? 'space30' : 'space0'}
+          >
+            {children}
+          </Text>
+        ) : null}
+        {direction === 'right' ? (
+          <Box
+            alignItems="center"
+            borderColor="inherit"
+            borderRadius="borderRadiusCircle"
+            borderStyle="solid"
+            borderWidth="borderWidth10"
+            display="flex"
+            height="sizeSquare80"
+            justifyContent="center"
+            width="sizeSquare80"
+          >
+            <ArrowForwardIcon decorative={false} display="inline-flex" title={nextArrowLabel} />
+          </Box>
+        ) : null}
+      </Box>
```

Comment:
This nested JSX is a bit on the wild side. I'd be tempted to pull them out into individual components. For example the boxes wrapping the arrow icons are identical.

I think I'd prefer to see the contents of the PaginationArrow JSX be a composition of styled components that just take children.

Reply:
I've refactored PaginationArrow. Let me know what you think.